sdk/src/main/scala/com/horizen/storage/BackupStorage.scala

```diff
@@ -0,0 +1,41 @@
+package com.horizen.storage
+
+import com.horizen.companion.SidechainBoxesCompanion
+import com.horizen.utils.{ByteArrayWrapper, Pair => JPair}
+
+import scala.util.Try
+import java.util.{ArrayList => JArrayList}
+import scorex.crypto.hash.Blake2b256
+
+class BackupStorage (storage : Storage, sidechainBoxesCompanion: SidechainBoxesCompanion) {
```

Comment:
`sidechainBoxesCompanion` is a constructor argument - it is good. But it is not used, I would expect it to be used in `update` for example or in `interator` wrapper.

Reply:
Yes the _sidechainBoxesCompaion_ is needed inside the **SidechainState** and **SidechainWallet** and as you suggested I moved it from an additional parameter to these classes to an internal member of the **BackupStorage** class.
Do you think is it good?

Reply:
ok, I will check it during the next CR session to see the full context. Thanks!